### PR TITLE
Fixed texture documentation

### PR DESCRIPTION
**Pull request by Quaker762**
Squashed two commits, adding texture.cpp and fixing documentation in
texture.hpp.

@z33ky  [Please no more haha](http://www.buyfruit.com.au/images/P/Spegetti-Squash.jpg)

### CODE REVIEW

File: include/SH3/graphics/texture.hpp [hunk 1]
```diff
@@ -1,56 +1,99 @@
...
+namespace sh3_graphics
+{
+    /**
+     * Full texture header.
+     *
+     * Both batch and individual texture.
+     */
+    typedef struct
+    {
+        std::uint32_t batchHeaderMarker;          /** This should be 0xFFFFFFFF to mark a header chunk */
```
Comment (z33ky) [inline include/SH3/graphics/texture.hpp]: Still needs `/**<`-comments.

File: include/SH3/graphics/texture.hpp [hunk 2]
```diff
@@ -1,56 +1,99 @@
...
-        all of the file types, including all of the textures, meshes and even the motion capture
-        skeletal animation that Konami captured. Thanks mate!
+    /**
+     *
+     * Describes a logical texture that can be bound to OpenGL
+     *
+     * Defines a few bits of data and some functions to load in a texture
+     * from a SILENT HILL 3 .arc section.both batch and individual texture
+     *
+     * @note
```
Comment (z33ky) [inline include/SH3/graphics/texture.hpp]: ?

File: include/SH3/graphics/texture.hpp [hunk 3]
```diff
@@ -1,56 +1,99 @@
...
+        texture(const std::string& filename, sh3_arc& arc){Load(filename, arc);}
+        ~texture(){}
 
---*/
-#ifndef SH3_TEXTURE_H_INCLUDED
-#define SH3_TEXTURE_H_INCLUDED
+        /**
+         * Load a texture and return a result to it
+         *
+         * @note
```
Comment (z33ky) [inline include/SH3/graphics/texture.hpp]: ?

**Review by z33ky — CHANGES_REQUESTED** [review]
Sorry.

Btw you don't need to open a new pull request after squashing. Just like adding new commits, Github can deal with changing commits as well.

File: include/SH3/graphics/texture.hpp (hunk 3 above)
Comment (Quaker762) [reply include/SH3/graphics/texture.hpp]: Oh sorry I forgot to answer this. I set up a few abbreviations in my IDE that includes the @note tag. It's a bit of bad habit but I can remove it.

File: include/SH3/graphics/texture.hpp (hunk 1 above)
Comment (Quaker762) [reply include/SH3/graphics/texture.hpp]: Oops, missed that `<`

**Pull request merged**